sdk/nodejs/tests/provider/experimental/analyzer.spec.ts

```diff
@@ -302,7 +303,32 @@ describe("Analyzer", function () {
     it("errors nicely for resource references", async function () {
         const dir = path.join(__dirname, "testdata", "resource-reference");
         const analyzer = new Analyzer(dir, "provider");
-        assert.throws(() => analyzer.analyze(), /Resource references are not supported yet: found type 'MyResource'/);
+        assert.throws(
+            () => analyzer.analyze(),
+            /Resource references are not supported yet: component 'MyComponent' property 'MyComponentArgs.aResource' has type 'MyResource'/,
+        );
+    });
+
+    it("errors nicely for invalid property types for top-level properties", async function () {
+        const dir = path.join(__dirname, "testdata", "bad-property-type", "top-level");
+        const analyzer = new Analyzer(dir, "provider");
+        assert.throws(
+            () => analyzer.analyze(),
+            (err) =>
+                err.message ===
+                "Union types are not supported: component 'MyComponent' input 'MyComponentArgs.invalidProp' has type 'string | boolean'",
+        );
+    });
+
+    it("errors nicely for invalid property types for sub-type properties", async function () {
+        const dir = path.join(__dirname, "testdata", "bad-property-type", "sub-type");
+        const analyzer = new Analyzer(dir, "provider");
+        assert.throws(
+            () => analyzer.analyze(),
+            (err) =>
+                err.message ===
+                "Unsupported type: component 'MyComponent' input 'MyOtherArgs.invalidProp[key]' has type '\"fixed value\"'",
```

Comment:
```suggestion
                "Unsupported type: component 'MyComponent' input map 'MyOtherArgs.invalidProp' has value type '\"fixed value\"'",
```

Something like this would maybe make it clearer, but it's more awkward to implement.

Reply:
I changed it to a slightly better

> Unsupported type for component 'MyComponent' input 'MyOtherArgs.invalidProp values': type '\"fixed value\"

Is `values` inside quotes too meh?

Reply:
It's a bit meh, but probably fine for now.